packages/flutter_tools/test/general.shard/fuchsia/fuchsia_ffx_test.dart

```diff
@@ -0,0 +1,136 @@
+      final FuchsiaFfx fuchsiaFfx = FuchsiaFfx(
+        fuchsiaArtifacts: mockFuchsiaArtifacts,
+        logger: logger,
+        processManager: FakeProcessManager.any(),
+      );
+
+      expect(() async => await fuchsiaFfx.list(),
+          throwsToolExit(message: 'Fuchsia ffx tool not found.'));
+    });
+
+    testWithoutContext('no device found', () async {
+      ffx.createSync();
+
+      final ProcessManager processManager =
+          FakeProcessManager.list(<FakeCommand>[
+        FakeCommand(
+          command: <String>[ffx.path, 'target', 'list', '--format', 's'],
+          exitCode: 0,
+          stderr: 'No devices found.',
+        ),
+      ]);
+
+      final FuchsiaFfx fuchsiaFfx = FuchsiaFfx(
+        fuchsiaArtifacts: mockFuchsiaArtifacts,
+        logger: logger,
+        processManager: processManager,
+      );
+
+      expect(await fuchsiaFfx.list(), isNull);
+      expect(logger.errorText, isEmpty);
+    });
+
+    testWithoutContext('error', () async {
+      ffx.createSync();
+
+      final ProcessManager processManager =
+          FakeProcessManager.list(<FakeCommand>[
+        FakeCommand(
+          command: <String>[ffx.path, 'target', 'list', '--format', 's'],
+          exitCode: 1,
+          stderr: 'unexpected error',
+        ),
+      ]);
+
+      final FuchsiaFfx fuchsiaFfx = FuchsiaFfx(
+        fuchsiaArtifacts: mockFuchsiaArtifacts,
+        logger: logger,
+        processManager: processManager,
+      );
+
+      expect(await fuchsiaFfx.list(), isNull);
+      expect(logger.errorText, contains('unexpected error'));
+    });
+
+    testWithoutContext('devices found', () async {
+      ffx.createSync();
+
+      final ProcessManager processManager =
+          FakeProcessManager.list(<FakeCommand>[
+        FakeCommand(
+          command: <String>[ffx.path, 'target', 'list', '--format', 's'],
+          exitCode: 0,
+          stdout: 'device1\ndevice2',
+        ),
+      ]);
+
+      final FuchsiaFfx fuchsiaFfx = FuchsiaFfx(
+        fuchsiaArtifacts: mockFuchsiaArtifacts,
+        logger: logger,
+        processManager: processManager,
+      );
+
+      expect(await fuchsiaFfx.list(), <String>['device1', 'device2']);
+      expect(logger.errorText, isEmpty);
+    });
+
+    testWithoutContext('timeout', () async {
+      ffx.createSync();
+
+      final ProcessManager processManager =
+          FakeProcessManager.list(<FakeCommand>[
+        FakeCommand(
+          command: <String>[ffx.path, '-T', '2', 'target', 'list', '--format', 's'],
+          exitCode: 0,
+          stdout: 'device1',
+        ),
+      ]);
+
+      final FuchsiaFfx fuchsiaFfx = FuchsiaFfx(
+        fuchsiaArtifacts: mockFuchsiaArtifacts,
+        logger: logger,
+        processManager: processManager,
+      );
+
+      expect(await fuchsiaFfx.list(timeout: const Duration(seconds: 2)),
+          <String>['device1']);
+    });
+  });
+}
+
+class MockFuchsiaArtifacts extends Mock implements FuchsiaArtifacts {}
```

Comment:
I would set this up as a Fake instead. We're trying to remove all usage of mockito to unblock our eventual null safe migration. For example:

```
class FakeFuchsiaArtifacts extends Fake implements FuchsiaArtifacts {
  @override
  File ffx;
}
```